src/bucket/Bucket.cpp

```diff
@@ -42,19 +306,39 @@ Bucket::Bucket(std::string const& filename, Hash const& hash)
                    mFilename);
         mSize = fs::size(filename);
     }
+
+    if (pageSize)
+    {
+        // Use individual key indexing if size if below cutoff
+        initIndex(mSize < indexCutoff ? 0 : *pageSize);
```

Comment:
I don't this this will be too bad, I was playing with separating the bucket and index more anyway. Probably the path of least resistance is building the index during merge and passing a unique_ptr of the constructed index to adoptFileAsBucket.

However, this does raise a question about persisting indexes on disk. Should we persist indexes on disk and support resuming a partial index like we resume partial merges? I haven't looked into it yet, but thinking through it I think it might be necessary if we want to do the indexing while we merge.